File: angular_fichajesPi/src/app/intranet/incidencias/model/incidencia.ts
import { Empleado } from "../../empleados/model/empleado"

export class Incidencia {

  id?: number
  resumen: string
  dia: string
  explicacion: string
  resuelta: boolean | null
  usuario: Empleado | null

  constructor(
    resumen: string,
    dia: string,
    explicacion: string,
    resuelta: boolean | null,
    usuario: Empleado | null
  ) {
    this.resumen = resumen;
    this.dia = dia;
    this.explicacion = explicacion;
    this.resuelta = resuelta;
    this.usuario = usuario;
  }

}

File: spring-boot_fichajesPi/src/main/java/com/fichajespi/entity/Incidencia.java
package com.fichajespi.entity;

import java.time.LocalDate;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.sun.istack.NotNull;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "incidencias")
public class Incidencia {

	@Id
	@GeneratedValue
	private Long id;

	@JsonIgnore
	@NotNull
	private String descripcion;

	@NotNull
	private String resumen;

//	@NotNull
//	@Temporal(TemporalType.DATE)
//	private Date dia;
	@NotNull
	@JsonFormat(pattern = "yyyy-MM-dd")
	private LocalDate dia;

	@Column(columnDefinition = "boolean default false")
	private Boolean resuelta;

	private String explicacion;

	@JsonIgnoreProperties(value = { "password", "dni", "diasVacaciones", "horasGeneradas",
			"working", "enVacaciones", "deBaja",
			"admin", "roles", "fichajes", "incidencias", "permisos", "vacaciones" })
	@ManyToOne
	@JoinColumn(name = "usuario_id")
	private Usuario usuario;

}
